### PR TITLE
feat: currency exchanger

### PR DESCRIPTION
**Pull request by paul-nguyen-goldenowl**
## Description
- fetch and cache latest exchange rate with [`https://api.apilayer.com/exchangerates_data`](https://api.apilayer.com/exchangerates_data)
- Exchange currency

### CODE REVIEW

File: GoMoney/Scences/Exchange/CountryPickerViewController.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+}
+
+extension CountryPickerViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let country = countries[indexPath.row]
+        let cell = UITableViewCell()
+        cell.textLabel?.text = country.country
+        cell.accessoryView = UIImageView(image: UIImage(named: country.country))
+        cell.accessoryView?.bounds = CGRect(x: 0, y: 0, width: 30, height: 20)
```
Comment (bobby-nguyen-goldenowl) [inline GoMoney/Scences/Exchange/CountryPickerViewController.swift:47]: should passing data into cell and handle in this
just simple call in this like: `cell.bindData(country)`

File: GoMoney/Scences/Exchange/ExchangeCell.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+class ExchangeCell: UITableViewCell {
+    static let identifier = "exchange_cell"
+    lazy var symbolLabel = GMLabel(style: .largeBold, isCenter: true)
+    lazy var codeLabel = GMLabel(isCenter: true)
+    lazy var rateLabel = GMLabel()
+    lazy var amountLabel = GMLabel(style: .largeBold)
+
+    lazy var flag: UIImageView = .build { _ in }
+
```
Comment (bobby-nguyen-goldenowl) [inline GoMoney/Scences/Exchange/ExchangeCell.swift:11]:
```suggestion
    lazy var flag: UIImageView = UIImageView()

```

File: GoMoney/Scences/Exchange/ExchangeViewController.swift [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            top: view.safeAreaLayoutGuide.topAnchor,
+            left: view.leftAnchor,
+            right: view.rightAnchor,
+            paddingTop: 16,
+            paddingLeft: 16,
+            paddingRight: 16,
+            height: 65
+        )
+
+        tableView.anchor(top: textField.bottomAnchor, left: view.leftAnchor, bottom: view.safeAreaLayoutGuide.bottomAnchor, right: view.rightAnchor, paddingTop: 16, paddingLeft: 16, paddingBottom: 16, paddingRight: 16)
```
Comment (bobby-nguyen-goldenowl) [inline GoMoney/Scences/Exchange/ExchangeViewController.swift:103]: break line

File: GoMoney/Service/ExchangeService.swift [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    let cache = URLCache.shared
+
+    func getExchangeRate(from base: String, to: [CurrencyUnit], completion: @escaping ExchangeCompletion) {
+        guard let url = URL(string: String(format: "%@?base=%@", ServiceResources.exchangeUri, base)) else { return }
+
+        print("url=\(url)")
+
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        request.addValue("3NhHH9XmhDpxPSmVcPKU71r3S5BkDDX2", forHTTPHeaderField: "apikey")
```
Comment (bobby-nguyen-goldenowl) [inline GoMoney/Service/ExchangeService.swift:31]: create a class Manager (or struct, enum,...) to manage apiKey

File: GoMoney/Scences/Exchange/ExchangeCell.swift (hunk 2 above)
Comment (paul-nguyen-goldenowl) [reply GoMoney/Scences/Exchange/ExchangeCell.swift:11]: I use `.build` to set `translatesAutoresizingMaskIntoConstraints = false`

File: GoMoney/Scences/Exchange/CountryPickerViewController.swift [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        let country = countries[indexPath.row]
+        onSelectCountry?(country)
+        dismiss(animated: true)
+    }
+}
+
+private extension UITableViewCell {
+    func bindData(country: CurrencyItem) {
```
Comment (bobby-nguyen-goldenowl) [inline GoMoney/Scences/Exchange/CountryPickerViewController.swift:65]: it should be extensioon for `ExchangeCell` , not `UITabbleVieweCell`

File: GoMoney/Scences/Exchange/CountryPickerViewController.swift [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+extension CountryPickerViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let country = countries[indexPath.row]
+        let cell = UITableViewCell()
+        cell.bindData(country: country)
+//        cell.textLabel?.text = country.country
+//        cell.accessoryView = UIImageView(image: UIImage(named: country.country))
+//        cell.accessoryView?.bounds = CGRect(x: 0, y: 0, width: 30, height: 20)
```
Comment (bobby-nguyen-goldenowl) [inline GoMoney/Scences/Exchange/CountryPickerViewController.swift:48]: remove old code

**Review by bobby-nguyen-goldenowl — APPROVED** [review]

**Pull request merged**